src/core/secondaryAdapters/keycloakOidcClient.ts

```diff
@@ -72,6 +72,8 @@ export async function createKeycloakOidcClient(params: {
     const oidcClient = id<OidcClient.LoggedIn>({
         "isUserLoggedIn": true,
         "accessToken": keycloakInstance.token!,
+        "expirationTime":
+            jwtSimple.decode(keycloakInstance.token!, "", true)["exp"] || "",
```

Comment:
This is not correct typewise.  
There's no type error because `jstSimple.decode` returns `any`.  
It means you you should be extra carefull.  

About `keycloakInstance.token!`:  
We ban the use of `!`.  Each time you assert something is not null, use:  
```typescript 
import { assert } from "tsafe/assert";

assert( x !== undefined);
``` 

This way if ever you where wrong in your assertion it will be much more reasy to debug because we'll get an error tracing back to the wrong assertion.  

Also, are you sure that the `exp` is in millisecond?  
Every token named `xxxTime: number` should be in millisecond.

Reply:
exp is in second. It's seems to be an epoch

Reply:
@smith-julien This has not been delt with.